easybuild/easyblocks/x/xalt.py

```diff
@@ -0,0 +1,160 @@
+            error_msg += "You can edit the easyconfig file, or use 'eb --try-amend=config_py=<path>'. "
+            error_msg += "See https://xalt.readthedocs.io/en/latest/030_site_filtering.html for more information."
+            raise EasyBuildError(error_msg)
+
+        # XALT system name is mandatory
+        if self.cfg['syshost']:
+            self.cfg.update('configopts', '--with-syshostConfig=%s' % self.cfg['syshost'])
+        else:
+            error_msg = "The name of the system must be specified via the 'syshost' easyconfig parameter. "
+            error_msg += "You can edit the easyconfig file, or use 'eb --try-amend=syshost=<string>'. "
+            error_msg += "See https://xalt.readthedocs.io/en/latest/020_site_configuration.html for more information."
+            raise EasyBuildError(error_msg)
+
+        # Transmission method is mandatory
+        if self.cfg['transmission']:
+            self.cfg.update('configopts', '--with-transmission=%s' % self.cfg['transmission'])
+        else:
+            error_msg = "The XALT transmission method must be specified via the 'transmission' easyconfig parameter. "
+            error_msg = "You can edit the easyconfig file, or use 'eb --try-amend=transmission=<string>'. "
+            error_msg += "See https://xalt.readthedocs.io/en/latest/020_site_configuration.html for more information."
+            raise EasyBuildError(error_msg)
+
+        # GPU tracking
+        if self.cfg['gpu_tracking'] is True:
+            # User enabled
+            self.cfg.update('configopts', '--with-trackGPU=yes')
+        elif self.cfg['gpu_tracking'] is None:
+            # Default value, enable GPU tracking if nvml.h is present
+            # and the CUDA module is loaded
+            cuda_root = get_software_root('CUDA')
+            if not cuda_root:
+                # Try using the system CUDA
+                cuda_root = '/usr/local/cuda'
+            nvml_h = os.path.join(cuda_root, "include", "nvml.h")
+            if os.path.isfile(nvml_h):
+                self.cfg.update('configopts', '--with-trackGPU=yes')
+                self.cfg['gpu_tracking'] = True
+        else:
+            # User disabled
+            self.cfg.update('configopts', '--with-trackGPU=no')
+
+        # MySQL
+        if self.cfg['mysql'] is True:
+            self.cfg.update('configopts', '--with-MySQL=yes')
+        else:
+            self.cfg.update('configopts', '--with-MySQL=no')
+
+        # Configure
+        super(EB_XALT, self).configure_step()
+
+    def make_module_extra(self, *args, **kwargs):
+        txt = super(EB_XALT, self).make_module_extra(*args, **kwargs)
+
+        txt += self.module_generator.prepend_paths('LD_PRELOAD', 'lib64/libxalt_init.so')
+        txt += self.module_generator.prepend_paths('SINGULARITY_BINDPATH', '')
+        txt += self.module_generator.prepend_paths('SINGULARITYENV_LD_PRELOAD', 'lib64/libxalt_init.so')
```

Comment:
Do these really belong in the `XALT` easyblock?

At the very least they should be done in a separate code block below, with an accompanying comment to clarify why it makes sense to let the `XALT` module file define update these `$SINGULARITY*` environment variables.

Reply:
The suggestion to add these variables comes from the [XALT documentation](https://xalt.readthedocs.io/en/latest/050_install_and_test.html#xalt-modulefile)